src/Webserver.API/Services/WebApp/ApiResourceHandler.cs

```diff
@@ -0,0 +1,186 @@
+using Siemens.Simatic.S7.Webserver.API.Models;
+using Siemens.Simatic.S7.Webserver.API.Services.RequestHandling;
+using Siemens.Simatic.S7.Webserver.API.StaticHelpers;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Siemens.Simatic.S7.Webserver.API.Services.WebApp
+{
+    /// <summary>
+    /// The ResourceHandler implements comfort functions to Deploy a single Resource or Download a WebApp Resource
+    /// </summary>
+    public class ApiResourceHandler : IApiResourceHandler
+    {
+        private readonly IApiRequestHandler ApiRequestHandler;
+
+        private readonly IApiWebAppResourceBuilder ApiWebAppResourceBuilder;
+
+        /// <summary>
+        /// The ResourceHandler implements comfort functions to Deploy a single Resource or Download a WebApp Resource
+        /// </summary>
+        /// <param name="apiRequestHandler"></param>
+        /// <param name="apiWebAppResourceBuilder"></param>
+        public ApiResourceHandler(IApiRequestHandler apiRequestHandler, IApiWebAppResourceBuilder apiWebAppResourceBuilder)
+        {
+            this.ApiRequestHandler = apiRequestHandler;
+            this.ApiWebAppResourceBuilder = apiWebAppResourceBuilder;
+        }
+
+        /// <summary>
+        /// make sure to set the webapp PathToWebAppDirectory before calling this method!
+        /// Will send a Createresource, Uploadticket and Closeticket request to the API
+        /// </summary>
+        /// <param name="webApp">make sure to set the webapp PathToWebAppDirectory before calling this method!</param>
+        /// <param name="resource">the resource you want to deploy - MUST have information:
+        /// Name
+        /// Media_type
+        /// Last_modified
+        /// optionally:</param>
+        /// etag, visibility
+        /// <returns></returns>
+        public async Task DeployResourceAsync(ApiWebAppData webApp, ApiWebAppResource resource)
+        {
+            string path = webApp.PathToWebAppDirectory + @"\" + resource.Name.Replace("/", "\\");
+            if (!File.Exists(path))
+                throw new FileNotFoundException($"file at: {path} has not been found - did you set the webApp PathToWebAppDirectory correctly? given: {Environment.NewLine + webApp.PathToWebAppDirectory}");
+            var ticketIdResponse = await ApiRequestHandler.WebAppCreateResourceAsync(webApp.Name, resource.Name, resource.Media_type, resource.Last_modified.ToString(DateTimeFormatting.ApiDateTimeFormat), resource.Visibility, resource.Etag);
+            string ticketId = ticketIdResponse.Result;
+            try
+            {
+                await ApiRequestHandler.UploadTicketAsync(ticketId, path);
+            }
+            catch (ApiTicketingEndpointUploadException ex)
+            {
+                await ApiRequestHandler.ApiCloseTicketAsync(ticketId);
+                throw ex;
```

Comment:
With this the stacktrace will not be changed
```suggestion
                throw;
```

Reply:
done